### ISSUE TITLE
RobertaForMaskedLM loss calculated wrong(?)

### ISSUE THREAD

**Issue by leo-liuzy** [huggingface/transformers#13020]
## Environment info

- `transformers` version: 4.8.1 (or github main as of 2021.8.5)
- Platform: MacOS (or any)
- Python version: 3.9 (or any 3.x)
- PyTorch version (GPU?): 1.9.0 cpu
- Tensorflow version (GPU?):
- Using GPU in script?: No
- Using distributed or parallel set-up in script?: No

## Information

Model I am using (Bert, XLNet ...): XLM-R/Roberta

The problem arises when using:
* [x] the official example scripts: (give details below)
* [ ] my own modified scripts: (give details below)

The tasks I am working on is:
* [x] an official GLUE/SQUaD task: (give the name) MaskedLM
* [ ] my own task or dataset: (give details below)

## To reproduce

1. [Tutorial example here](https://huggingface.co/transformers/model_doc/xlmroberta.html#transformers.XLMRobertaForMaskedLM)
2. [Source code here](https://github.com/huggingface/transformers/blob/60e448c87eff29b166bf2821f5389056a72343e3/src/transformers/models/roberta/modeling_roberta.py#L1105)
3.

## Expected behavior
The logic of calculating MLM seems wrong. Shouldn't CrossEntropyLoss only run on **masked tokens** rather than all tokens? I see no operation here.

**Comment by NielsRogge** [comment]
The calculation is correct. However, it's the responsibility of the user to prepare the labels, so you need to make sure you set the labels to -100 for positions where you don't want to incur a loss (as -100 is the `ignore_index` of PyTorch's `CrossEntropyLoss`).

**Comment by leo-liuzy** [comment]
Ah, that makes sense, thanks!. I think you should modify the tutorial page in documentation page? Or put what you said somewhere in the doc pagr?
